### PR TITLE
make assert soft

### PR DESCRIPTION
**Pull request by emord**
@czue

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 1]
```diff
@@ -604,7 +604,9 @@ def _remove_case(to_remove):
                 # if the case had indexes they better also be in our removal list (except for ourselves)
                 for index in indices.values():
                     if not _domain_has_legacy_toggle_set():
-                        assert index in candidates_to_remove, \
+                        # unblocking http://manage.dimagi.com/default.asp?185850#1039475
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: fogbugz url anchors make me cringe....

**Comment by czue** [comment]
:+1: when tests pass (seems overly paranoid but it's critical code so will wait)

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 2]
```diff
@@ -604,8 +604,10 @@ def _remove_case(to_remove):
                 # if the case had indexes they better also be in our removal list (except for ourselves)
                 for index in indices.values():
                     if not _domain_has_legacy_toggle_set():
-                        assert index in candidates_to_remove, \
-                            "expected {} in {} but wasn't".format(index, candidates_to_remove)
+                        # unblocking http://manage.dimagi.com/default.asp?185850#1039475
+                        _assert = soft_assert(to=['czue' + '@' + 'dimagi.com'], exponential_backoff=True)
```
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: fyi, `exponential_backoff` is `True` by default

**Comment by proteusvacuum** [comment]
is there a test case for when this assert gets hit?

**Comment by czue** [comment]
@proteusvacuum i don't think so. just happened to fail because it was a syntax error

**Comment by proteusvacuum** [comment]
was just wondering if there was a specific instance when `index` wasn't in `candidates_to_remove` (mostly just curious what would cause this assert to fire)

**Comment by czue** [comment]
[you and me both](http://manage.dimagi.com/default.asp?185850).

**Comment by proteusvacuum** [comment]
gnarly.

**Pull request merged**